src/js/components/FilterTree.js

```diff
@@ -15,113 +15,97 @@ import Current from "../state/Current"
       ""
     )
 
-    function onNodeClick() {
-      dispatch(
-        SearchBar.restoreSearchBar({
-          pinned: getPinnedFilters(node),
-          current: node.data.filter,
-          previous: node.data.filter,
-          error: null,
-          editing: null
-        })
-      )
-      const nodeSpaceId = get(
-        node,
-        ["data", "finding", "search", "spaceId"],
-        ""
-      )
-      dispatch(Current.setSpaceId(nodeSpaceId))
-      dispatch(submitSearch({history: false, investigation: false}))
-    }
-
-    let className = classNames("filter-tree-node", {
-      pinned: nodeIsPinned(pinnedFilters, node),
-      active: nodeIsActive(pinnedFilters, previous, node)
-    })
-
-    function renderWarning() {
-      const findingSpaceId = get(
-        node,
-        ["data", "finding", "search", "spaceId"],
-        ""
-      )
-
-      const tip = "This space no longer exists"
-      if (includes(spaceIds, findingSpaceId)) return null
-
-      return (
-        <div
-          className="warning-body"
-          data-tip={tip}
-          data-effect="solid"
-          data-place="right"
-        >
-          <Warning />
-          <ReactTooltip />
-        </div>
-      )
-    }
-
-    const template = [
-      {
-        label: "Delete",
-        click: () => {
-          let multiTs = node.mapChildren((node) => node.data.finding.ts)
-          globalDispatch(Investigation.deleteFindingByTs(multiTs))
-        }
-      },
-      {type: "separator"},
-      {
-        label: "Delete All",
-        click: () => globalDispatch(Investigation.clearInvestigation())
-      }
-    ]
-
-    const openMenu = usePopupMenu(template)
-    const onContextMenu = () => {
-      openMenu()
-    }
+    const tip = "This space no longer exists"
+    if (includes(spaceIds, findingSpaceId)) return null
 
     return (
-      <div key={i} className={className}>
-        <div
-          className="filter-tree-parent"
-          onClick={onNodeClick}
-          onContextMenu={onContextMenu}
-          title={findingSpaceName}
-        >
-          <FilterNode filter={node.data.filter} />
-          {renderWarning()}
-        </div>
-        <div className="filter-tree-children">
-          {node.children.map(renderNode)}
-        </div>
+      <div
+        className="warning-body"
+        data-tip={tip}
+        data-effect="solid"
+        data-place="right"
+      >
+        <Warning />
+        <ReactTooltip />
       </div>
     )
   }
 
-  let tree = createInvestigationTree(investigation)
+  return (
+    <div key={i} className={className}>
+      <div
+        className="filter-tree-parent"
+        onClick={onNodeClick}
+        onContextMenu={() => menu.open()}
+        title={findingSpaceName}
+      >
+        <FilterNode filter={node.data.filter} />
+        {renderWarning()}
+      </div>
+      <div className="filter-tree-children">
+        {node.children.map((node, i) => (
+          <NodeRow node={node} i={i} key={i} />
```

Comment:
The NodeRow component used to be a plain function. I converted it to Components so that the hooks in them will be associated with each Row, rather than the parent component.